aws-rds-integration/aws-rds-integration.json

```diff
@@ -0,0 +1,166 @@
+  "sourceUrl": "https://github.com/aws-cloudformation/aws-cloudformation-rpdk.git",
+  "properties": {
+    "IntegrationName": {
+      "description": "The name of the integration.",
+      "type": "string",
+      "minLength": 1,
+      "maxLength": 64
+    },
+    "Tags": {
+      "type": "array",
+      "maxItems": 50,
+      "uniqueItems": true,
+      "insertionOrder": false,
+      "description": "An array of key-value pairs to apply to this resource.",
+      "items": {
+        "$ref": "#/definitions/Tag"
+      }
+    },
+    "SourceArn": {
+      "type": "string",
+      "description": "The Amazon Resource Name (ARN) of the Aurora DB cluster to use as the source for replication."
+    },
+    "TargetArn": {
+      "type": "string",
+      "description": "The ARN of the Redshift data warehouse to use as the target for replication."
+    },
+    "IntegrationArn": {
+      "type": "string",
+      "description": "The ARN of the integration."
+    },
+    "KMSKeyId": {
+      "type": "string",
+      "description": "The AWS Key Management System (AWS KMS) key identifier for the key used to to encrypt the integration. "
+    },
+    "AdditionalEncryptionContext": {
+      "type": "array",
+      "insertionOrder": false,
+      "items": {
+        "$ref": "#/definitions/EncryptionContextNameValuePair"
+      },
+      "description": "An optional set of non-secret key\u2013value pairs that contains additional contextual information about the data."
+    },
+    "CreateTime": {
+      "type": "string"
+    }
+  },
+  "required": [
+    "SourceArn",
+    "TargetArn",
+    "IntegrationName"
+  ],
+  "definitions": {
+    "Tags": {
+      "type": "array",
+      "maxItems": 50,
+      "uniqueItems": true,
+      "insertionOrder": false,
+      "description": "An array of key-value pairs to apply to this resource.",
+      "items": {
+        "$ref": "#/definitions/Tag"
+      }
+    },
+    "Tag": {
+      "description": "A key-value pair to associate with a resource.",
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "Key": {
+          "type": "string",
+          "description": "The key name of the tag. You can specify a value that is 1 to 128 Unicode characters in length and cannot be prefixed with aws:. You can use any of the following characters: the set of Unicode letters, digits, whitespace, _, ., /, =, +, and -. ",
+          "minLength": 1,
+          "maxLength": 128
+        },
+        "Value": {
+          "type": "string",
+          "description": "The value for the tag. You can specify a value that is 0 to 256 Unicode characters in length and cannot be prefixed with aws:. You can use any of the following characters: the set of Unicode letters, digits, whitespace, _, ., /, =, +, and -. ",
+          "minLength": 0,
+          "maxLength": 256
+        }
+      },
+      "required": [
+        "Key"
+      ]
+    },
+    "EncryptionContextNameValuePair": {
```

Comment:
Now that you mention it, I went to search for prior art on this, and I found a much better way to do it ([code](https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-workspaces-web/blob/44274f3720b1622fefb78fc243ebcc1260bedf78/aws-workspacesweb-portal/aws-workspacesweb-portal.json#L69)). I originally did this because I thought it's impossible to use an arbitrary key-value map but evidently it's possible with `patternProperties`! Now it should be much cleaner, please check.